### PR TITLE
[HYDRATOR-374] Move Mock APIs from Hydrator Plugins to CDAP

### PR DESCRIPTION
**Pull request by knazir**
Issue: https://issues.cask.co/browse/HYDRATOR-374
PR in CDAP: https://github.com/caskdata/cdap/pull/6610

Whenever there is an API change in CDAP, Hydrator Plugins will face compilation errors. This slows down development and leads to bad times in general. Moving the mock APIs used for Hydrator testing to the Hydrator-Test package in CDAP should resolve this issue as API changes leading to compilation errors will be picked up immediately when building CDAP.

### CODE REVIEW

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/KVTableSinkTest.java [hunk 1]
```diff
@@ -35,7 +39,8 @@ public void testTableSinkWithMissingKeyField() {
     KVTableSink.KVTableConfig kvTableConfig = new KVTableSink.KVTableConfig("purchases", "rowKey", "user");
     KVTableSink kvTableSink = new KVTableSink(kvTableConfig);
 
-    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(inputSchema);
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(inputSchema,
```
Comment (albertshau) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/KVTableSinkTest.java]: There is a constructor that just takes the input schema right? If so, just use that one.

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/KVTableSinkTest.java [hunk 2]
```diff
@@ -50,7 +55,10 @@ public void testTableSinkWithMissingValueField() {
...
+                                                                               ImmutableMap.<String, Object>of(
```
Comment (albertshau) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/KVTableSinkTest.java]: I don't imagine these KVTable tests require a validator. Should remove if not needed.

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/TableSinkTest.java [hunk 3]
```diff
@@ -45,7 +49,10 @@ public void testTableSinkWithOutputSchemaExtraField() {
     TableSinkConfig tableSinkConfig = new TableSinkConfig("tableSink", "rowkey", outputSchema.toString());
     TableSink tableSink = new TableSink(tableSinkConfig);
 
-    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(inputSchema);
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(inputSchema,
+                                                                               ImmutableMap.<String, Object>of(
+                                                                                 CoreValidator.ID,
```
Comment (albertshau) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/TableSinkTest.java]: similar comment, remove if not needed

**Comment by albertshau** [comment]
just a comment about removing the validator if its not needed. Should only be required for the validator tests. Otherwise, its good to finally have these removed!

**Comment by albertshau** [comment]
lgtm

**Comment by knazir** [comment]
This is waiting on BUT to finish so that Hydrator Plugins can compile with the moved APIs in CDAP.

**Comment by knazir** [comment]
Guild is breen: http://builds.cask.co/browse/HYP-BAD102/latest going to squash and merge.

**Pull request merged**